src/error.rs

```diff
@@ -21,6 +21,15 @@ pub enum ScannerError {
 
     #[error("Operation timed out")]
     Timeout,
+
+    #[error("Block number {0} exceeds latest block {1}")]
+    BlockExceedsLatest(u64, u64),
```

Comment:
```suggestion
    #[error("{0} {1} exceeds latest block {2}")]
    BlockExceedsLatest(&'static str, u64, u64),
```
Do you think it makes sense to include the field name that failed the validation? 
This would provide more context to the user :thinking: